Add DNN ROI SP chain as an option, set default to false

wjdanswjddl opened the pull request:
## Description 

This PR includes updates to enable DNN ROI finding, as well as configurations for generating samples for training the DNN. When DNN ROI is enabled, `simtpc2d:dnnsp` (or `sptpc2d:dnnsp` for data) `recob:Wire` products are saved, _in addition to_ the traditional `gauss` and `wiener` products. 

 For now, the option to run signal processing with DNN ROI is set to false, so this PR doesn't immediately affect the current workflow in any way. Using the DNN ROI correctly will require [wirecell release](https://cdcvs.fnal.gov/redmine/issues/29487) and [larwirecell PR#56](https://github.com/LArSoft/larwirecell/pull/56) -- without these updates, the DNN ROI will still run, but the outputs will be wrong.

Detailed notes on updates:

- `wcls-sim-drift-depoflux-nf-sp.jsonnet`, `wcls-nf-sp-data.jsonnet`, `wcls-nf-sp.jsonnet` were updated to run just the current SP chain using traditional ROI finding workflow by default (`use_dnnroi == false`) and run both traditional and DNN ROI finding SP chains when `use_dnnroi == true`.
- `wcsimsp_sbnd.fcl` and `wcsp_data_sbnd.fcl` fcls are updated with DNN ROI specific external variables:

    - `use_dnnroi`: option to run SP chain using DNN ROI finding, in addition to the traditional SP chain
    - `nchunks`: number of chunks to divide the wire dimension in for inference, must match the NN training configuration 
    - `tick_per_slice`: scale factor for downsampling in the time dimension for inference, must match the NN training configuration
    - `dnnroi_model_p0`, `dnnroi_model_p1`: path to NN file, to be stored in `/cvmfs/sbnd.opensciencegrid.org/products/sbnd/sbnd_data/<version>/WireCell`, which is on $WIRECELL_PATH. 

- `cafmakerjob_sbnd_sce_systtools_and_fluxwgt_dnnsp.fcl`,  `reco1_data_dnnsp.fcl`, `standard_reco1_sbnd_dnnsp.fcl` are fcls that run on the `dnnsp` product
- `wcls-sim-drift-depoflux-nf-sp-samples_{tru,rec}.jsonnet`, `wirecell_sim_{tru,rec}.fcl`: jsonnets and fcls for generating training samples

## Checklist
- [x] Added at least 1 label from [available labels](https://github.com/SBNSoftware/sbndcode/issues/labels?sort=name-asc).
- [x] Assigned at least 1 reviewer under `Reviewers`,
- [x] Assigned all contributers including yourself under `Assignees`
- [ ] Linked any relevant issues under `Developement`
- [ ] Does this PR affect CAF data format? If so, please assign a CAF maintainer ([PetrilloAtWork](https://github.com/PetrilloAtWork) or [JosiePaton](https://github.com/JosiePaton)) as additional reviewer.
- [x] Does this affect the standard workflow? (changes files in standard workflow, but does not take into effect immediately)

### Relevant PR links (optional)
Does this PR require merging another PR in a different repository (such as sbnanobj/sbnobj etc.)?

### Link(s) to docdb describing changes (optional)
Is there a docdb describing the issue this solves or the feature added?

lynnt20 commented:
picky comment @wjdanswjddl... looking at `wirecell_sim_rec.fcl` and `wirecell_sim_tru.fcl`, could you add updated description? Since the current description is based of detsim stage. 

I'm assuming these files will only be used by experts (mainly you), but since the configs are pretty redundant, you could just add a new block to `wirecellsimsp_sbnd.fcl` or create a new config fcl called something like `wirecellsimp_dnn_sbnd.fcl`.

wjdanswjddl commented:
@lynnt20 thanks for the quick review! The sample making files were nonsense due to a merge gone bad, I've cleaned them up now. I also updated suitable descriptions and moved the config files into `wcsimsp_sbnd.fcl` as config blocks per your suggestions.

lynnt20 commented on sbndcode/WireCell/cfg/pgrapher/experiment/sbnd/wcls-nf-sp-data.jsonnet:
```diff
@@ -124,6 +129,22 @@ local wcls_output = {
+      frame_scale: [0.02, 0.02, 0.02],
+      nticks: params.daq.nticks,
+
+      chanmaskmaps: ['dnnspbad'],
```
just as a comment, in order to run imaging on the output with dnnsp, I'm not sure that this `chanmaskmaps` config will output all of the bad channels, and we also will need to save the summaries. 

but I think we'll need more work to check output of dnnsp with imaging anyways, so we should leave it for a future PR!

lynnt20 commented on sbndcode/JobConfigurations/standard/reco/reco1_data_dnnsp.fcl:
```diff
@@ -0,0 +1,32 @@
+
+outputs.out1.SelectEvents: [ "reco1" ]
+
+physics.producers.gaushit.CalDataModuleLabel: "sptpc2d:dnnsp"
```
just to reiterate, since signal processing is run in the reco1 stage for data, you need to inherit the correct fcl block to run dnn instead of traditional here!

lynnt20 commented on sbndcode/JobConfigurations/standard/standard_reco1_sbnd_dnnsp.fcl:
bump, comments from the `reco1_data_dnnsp.fcl` file also apply to this one.

bear-is-asleep commented:
@wjdanswjddl can you address the review comments?

nathanielerowe commented:
@wjdanswjddl any follow up on this?

wjdanswjddl commented:
I'm closing this and reopening a new PR as I've rebased and refactored some configs (which will also address Lynn's previous comment)
